src/registrar.js

```diff
@@ -0,0 +1,165 @@
+'use strict'
+
+const assert = require('assert')
+const debug = require('debug')
+const log = debug('libp2p:peer-store')
+log.error = debug('libp2p:peer-store:error')
+const errCode = require('err-code')
+
+const { Connection } = require('libp2p-interfaces/src/connection')
+const PeerInfo = require('peer-info')
+const MulticodecTopology = require('./connection-manager/topology')
+
+/**
+ * Responsible for notifying registered protocols of events in the network.
+ */
+class Registrar {
+  /**
+   * @param {PeerStore} peerStore
+   * @constructor
+   */
+  constructor (peerStore) {
+    this.peerStore = peerStore
+
+    /**
+     * Map of connections per peer
+     * TODO: this should be handled by connectionManager
+     * @type {Map<string, conn>}
+     */
+    this.connections = new Map()
+
+    /**
+     * Map of topologies per multicodec
+     *
+     * @type {Map<string, object>}
+     */
+    this.multicodecTopologies = new Map()
```

Comment:
This should just be `topologies`, the registrar shouldn't care about multicodecs, it really shouldn't have any knowledge of the internals of a Topology. It should just care about its desired connection min/max values and the current peers the topology has identified as being in their desired topology.